### PR TITLE
Use proper crate organization.

### LINKED ISSUE #14: Organize code into lib
Here's a good reference for organizing a binary project with its own lib:

https://doc.rust-lang.org/book/second-edition/ch07-00-modules.html

### PR DESCRIPTION
**Pull request by DanielSank**
Fixes #14

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
+extern crate bit_vec;
+extern crate num;
+extern crate rand;
+extern crate rustty;
+
+use std::collections::{HashSet};
```
Comment (lovesegfault) [inline src/lib.rs]: Unnecessary {}

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+use bit_vec::BitVec;
+use num::bigint::BigInt;
+use rand::Rng;
+use rustty::ui::{Widget};
```
Comment (lovesegfault) [inline src/lib.rs]: Unnecessary {}

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+use rustty::{CellAccessor};
```
Comment (lovesegfault) [inline src/lib.rs]: Unnecessary {}

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    grid: CellSet,
+}
+
+impl World {
+    pub fn new((width, height): Cell, rule: BigInt) -> World {
+        let (_, bytes) = rule.to_bytes_be();
+        World {
+            height: height,
+            width: width,
+            rule: BitVec::from_bytes(&bytes),
```
Comment (lovesegfault) [inline src/lib.rs]: Would  `BitVec::from_bytes(&rule.to_bytes_be().1)` not work? Seems more concise.

File: src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        for x in 1..self.width {
+            for y in 1..self.height {
+                if rand::thread_rng().gen_weighted_bool(30) {
+                    self.grid.insert((x, y));
+                }
+            }
+        }
+    }
+
+    fn get_present_state(&self, cell: &Cell) -> usize {
```
Comment (lovesegfault) [inline src/lib.rs]: Why not just call it `get_state()`? Present state seems redundant in this context.

File: src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        if bot {
+            val += (self.grid.contains(&(x, y+1)) as usize) << 7;
+        }
+        if bot && right {
+            val += (self.grid.contains(&(x+1, y+1)) as usize) << 8;
+        }
+        val
+    }
+
+    fn make_dead_or_alive(&self, cell: &Cell) -> bool {
```
Comment (lovesegfault) [inline src/lib.rs]: We need a better name, maybe `decide_state()`?

File: src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+        let state = self.get_present_state(cell);
+        if state > self.rule.len() -1 {
+            return false
+        }
+        return self.rule[state];
+    }
+
+    // This is an obviously dumb way to do this
+    // TODO: Find a better way
+    fn neighbors(&self, cell: &Cell) -> CellSet {
```
Comment (lovesegfault) [inline src/lib.rs:93]: This function and `get_present_state()` have really similar logic, we should maybe work on joining them or at least adding a third function that they both call for neighbor validation.

**Review by lovesegfault — CHANGES_REQUESTED** [review]
Just some general fixes we ought to do now before merging into `master`.

File: src/lib.rs (hunk 4 above)
Comment (DanielSank) [reply src/lib.rs]: Yup done.

File: src/lib.rs (hunk 6 above)
Comment (DanielSank) [reply src/lib.rs]: Yeah, fixed.

File: src/lib.rs (hunk 5 above)
Comment (DanielSank) [reply src/lib.rs]: done

File: src/lib.rs (hunk 7 above)
Comment (DanielSank) [reply src/lib.rs:93]: I agree. Making this an issue.

File: README.md [hunk 8]
```diff
@@ -3,3 +3,10 @@
 
 A Cellular Automaton written in Rust
 
+To run, install rust and then do this
+
+```
+cargo run --release xxxxxxx
+```
+
+where `xxxxxxx` is a number that encodes the rule for the automaton. This number can be anything from 0 to 2^512 -1.
```
Comment (lovesegfault) [inline README.md:12]: Add instructions to install Rust

**Review by lovesegfault — APPROVED** [review]

**Pull request merged**